### ISSUE TITLE
Core::spawn should expect Error=Void not Error=()

### ISSUE THREAD

**Issue by Rufflewind** [tokio-rs/tokio-core#197]
Since `spawn` never looks at the results of the future, the return value ought to be completely devoid of any useful information to reduce the risk that the user might unintentionally forget about the result.

The current choice of `Future<Item=(), Error=()>` still has one bit of information: it’s possible to distinguish between `Ok(())` and `Err(())` results.  It really ought to return `Future<Item=(), Error=Void>`, which means the `Future` is guaranteed to succeed.

It’s not a big problem by any means, just kind of quirky in my perspective.

**Comment by alexcrichton** [comment]
Yes unfortunately there's no shared "void" type right now across these libraries. The `futures` crate for example has a number of situations where the error should be void vs `()` as it is today.

This is something that we'll definitely want to evaluate for futures 0.2

**Comment by vks** [comment]
What about using the [`void` crate](https://crates.io/crates/void)?

**Comment by carllerche** [comment]
I don't think we want to pull in another crate. Either way, changing the type will be a breaking change which requires a version bump.

I think at this point, we'll be better off waiting for upcoming changes to rust-lang to hit stable.

**Comment by srijs** [comment]
Just for reference, this is being discussed in `futures` as well: https://github.com/alexcrichton/futures-rs/issues/476

If `futures` ended up exporting `Void` I suppose `tokio` could leverage that?

**Comment by alexcrichton** [comment]
Indeed!
